app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt

```diff
@@ -146,6 +151,21 @@ class HomeActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelecte
             createShortcut()
         }
         nav_view.getHeaderView(0).login_button.setOnClickListener {
+            val builder = AlertDialog.Builder(this)
```

Comment:
why are you coding inside the login section. You have to delete everything on logout, and not when everytime a user clicks on the login button. check the specific condition of logout and code in that please

Reply:
` nav_view.getHeaderView(0).login_button.text = "Logout"`
clicklistener is applied only after text of login-button is LOGOUT.

Reply:
arnt you coding directly in the login clicklistener? rather coding in the condition specific to logout?

Reply:
please check the code, we are removing shortcuts and invalidating tokens on logout............ code in that block, this code is prone to fail

Reply:
when the loginbtntext is not logout then we already have a listener that starts an intent .
`  nav_view.getHeaderView(0).login_button.setOnClickListener {
            startActivity(intentFor<LoginActivity>().singleTop())
        }`

Reply:
@SaumyaSingh1 Oh wait, I get your point. The codebase has issues itself. Remove this condition as well 
`if (nav_view.getHeaderView(0).login_button.text == "Logout") `
it is useless

Reply:
@adityastic If condition `if (nav_view.getHeaderView(0).login_button.text == "Logout") `  will be removed then how will we reach condition specific to logout ? It's the only condition that takes us to the logout ..

Reply:
@aggarwalpulkit596 please check if the code is done in clicklistener of logout or not .

Reply:
We assign the text to logout just before the click listener is set. How does checking it again make sense?. Your code is correct, you can also remove this condition as it is redundant

Reply:
@adityastic  Thankyou for your reviews :)

Reply:
Use the custom dialog for confirmation

Reply:
`if (nav_view.getHeaderView(0).login_button.text == "Logout")` you can remove this condition it's useless